src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -164,14 +162,13 @@ int replicationFactor() {
     public void run() {
       try {
         TopicState topicState = topicState();
-        if (topicState == null) {
-          if (!_topicCreationEnabled) {
-            throw new RuntimeException(_serviceName + ": topic, " + _topic + ", does not exist and topic creation is not enabled.");
-          }
-
-          topicState = createTopic();
+        
+        if (topicState == null && !_topicCreationEnabled) {
+          throw new RuntimeException(_serviceName + ": topic, " + _topic + ", does not exist and topic creation is not enabled.");
         }
 
+        topicState = createTopic();
```

Comment:
createTopic() is not consistent with what it is doing. It seems more readable and concise to do `topicFactory.createTopicIfNotExist` here directly and remove `createTopic()` method.

I am also wondering whether topicState may be null after this point. If yes, then we will see NPE.

How about doing this:

```
if (_topicCreationEnabled)
  topicFactory.createTopicIfNotExist(...)

TopicState topicState = topicState();

if (topicState == null)
  throw new RuntimeException(...) 
```

I realized that we need `setUncaughtExceptionHandler` in all services to catch exception thrown out of `run()`. I can submit a patch to do it later.